### ISSUE TITLE
Errors when using default controller profiles

### ISSUE THREAD

**Issue by hridpath** [XRTK/XRTK-Core#26]
When testing Teleport cursor and Line Pointer using the Default Gizmo via a Samsung HMD, the pointer, gizmo and parabolic line will have excessive jittering and episodes of not rendering or complete an action. No additional scritps have been added.

**To Reproduce**

1. Create a new project.
2. Add latest version of XRTK master(02/13/2019)
    2.a On default scene, configure from Mixed Reality menu. 
3. Add a Unity Terrain object with gound texture
4. Run using Odyssey+

**Expected behavior**
Same navigation ability as before with teleport and strafe

![20190213_152506_mixedreality_moment](https://user-images.githubusercontent.com/15804010/52750582-e3d6e280-2fa9-11e9-9619-8c50c29c7304.jpg)

**Information(please complete the following information):**
 - Version: current
Latest Release]

**Comment by hridpath** [comment]
Additional information. Error Stacks
MissingReferenceException: The object of type 'LinePointer' has been destroyed but you are still trying to access it.
Your script should either check if it is null or you should not destroy the object.
XRTK.SDK.UX.Pointers.BaseControllerPointer.SetCursor (UnityEngine.GameObject newCursor) (at Assets/XRTK.SDK/Features/UX/Scripts/Pointers/BaseControllerPointer.cs:100)
XRTK.SDK.UX.Pointers.BaseControllerPointer+<Start>d__19.MoveNext () (at Assets/XRTK.SDK/Features/UX/Scripts/Pointers/BaseControllerPointer.cs:150)
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () (at <ab1dc506f380448587ff80868d46ed83>:0)
System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_0 (System.Object state) (at <ab1dc506f380448587ff80868d46ed83>:0)
UnityEngine.UnitySynchronizationContext+WorkRequest.Invoke () (at C:/buildslave/unity/build/Runtime/Export/Scripting/UnitySynchronizationContext.cs:112)
UnityEngine.UnitySynchronizationContext.Exec () (at C:/buildslave/unity/build/Runtime/Export/Scripting/UnitySynchronizationContext.cs:73)
System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <ab1dc506f380448587ff80868d46ed83>:0)
Rethrow as TargetInvocationException: Exception has been thrown by the target of an invocation.
System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <ab1dc506f380448587ff80868d46ed83>:0)
System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) (at <ab1dc506f380448587ff80868d46ed83>:0)
XRTK.Utilities.Async.Internal.SyncContextUtility.ExecuteContinuations () (at Assets/XRTK/Utilities/Async/Internal/SyncContextUtility.cs:52)
UnityEditor.EditorApplication.Internal_CallUpdateFunctions () (at C:/buildslave/unity/build/Editor/Mono/EditorApplication.cs:200)

MissingReferenceException: The object of type 'LinePointer' has been destroyed but you are still trying to access it.
Your script should either check if it is null or you should not destroy the object.
XRTK.SDK.UX.Pointers.BaseControllerPointer.SetCursor (UnityEngine.GameObject newCursor) (at Assets/XRTK.SDK/Features/UX/Scripts/Pointers/BaseControllerPointer.cs:100)
XRTK.SDK.UX.Pointers.BaseControllerPointer+<Start>d__19.MoveNext () (at Assets/XRTK.SDK/Features/UX/Scripts/Pointers/BaseControllerPointer.cs:150)
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () (at <ab1dc506f380448587ff80868d46ed83>:0)
System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_0 (System.Object state) (at <ab1dc506f380448587ff80868d46ed83>:0)
UnityEngine.UnitySynchronizationContext+WorkRequest.Invoke () (at C:/buildslave/unity/build/Runtime/Export/Scripting/UnitySynchronizationContext.cs:112)
UnityEngine.UnitySynchronizationContext.Exec () (at C:/buildslave/unity/build/Runtime/Export/Scripting/UnitySynchronizationContext.cs:73)
UnityEngine.UnitySynchronizationContext.ExecuteTasks () (at C:/buildslave/unity/build/Runtime/Export/Scripting/UnitySynchronizationContext.cs:94)

**Comment by wassx** [comment]
Can you upload your example project please?

**Comment by SimonDarksideJ** [comment]
I think @hridpath is suggesting this is in the default setup.  So let's try using the 3D scene template that comes with Unity @wassx and see if we can replicate it.
I've not seen any such jitteryness but I have see the LinePointer error, however I think they are unrelated.

**Comment by hridpath** [comment]
@wassx , @SimonDarksideJ is correct. I am using the default profiles. I am testing navigation using the WHMD and Motion Controller. 
The interaction behaves normally for the first few teleports but then the onscreen control will freeze for up to 5 sec. and comes back but the pointer line can be multiplied (to me this indicates a jumping endpoint) and when teleporting, the target will jump around and the teleport parabolic line will seems to chase the target. 
When this is happening the teleport visual will stay on screen even after I have released the thumb stick. When it clears, I the camera will move but not always
 
I have a 138k tris and 147 verts, with a frame rate of 65-85 fps. when running via the editor.

Let me know if you have anymore questions.

**Comment by hridpath** [comment]
@wassx , @SimonDarksideJ .... **This issue may not be associated with the XRTK**. I am finding that I am getting similar behaviors for the controller and WHMD in the Mixed Reality Home. 

I am investigating more. I will be uninstalling the drivers for the Odyssey + and let the system find the latest.

**Comment by StephenHodgson** [comment]
Hey guys, sorry about the late replies, I've been pretty swamped lately at work.

I'll be looking into this asap. If you find any solutions please open a pull request as well.

**Comment by hridpath** [comment]
UPDATE: @StephenHodgson 
 I have changed the HMD for the original Odyssey as well as controllers.
 I am on Windows 10 v 1809 build 17763.316. 
I have uninstalled Windows Mixed Reality -restarted the hardware- connected HMD and completed installation of WMR and Portal.
I have tweeked my video card and my processor to get the best performance I can.

The issue is still present. Even doing the intro training for the WMR portal, the controllers continually gets lost, locks-up, disappears. This is the same behavior that caused me to create this ticket. 

I am using an Asus BT USB dongle. Does anyone think that this BT device may be the issue? 
Has anyone been able to reproduce the above described behaviors?
And the last question, could this be a side effect from the latest MS Security updates?

**Comment by SimonDarksideJ** [comment]
Then it sounds like an issue with your MR rig and not XRTK.

MR is particular about the type of Bluetooth adapter, so if your headset is moving fine and it's just the controllers that are jittery, then it could be your BT adapter causing your issues.
This reddit post is one of the best discussions around the best adapters to use, need to be 4 or 4.1 for the best fit.

The MS post also gives some advice on the best hardware to use.

The new Samsung Odyssey + includes BT built in to the headset, which is a nice touch, instead of requiring a new adapter :S

**Comment by hridpath** [comment]
I am closing this issue. Thanks to your suggestions, I realized that the contorllers were connecting as duplicates because I inadvertentlly use the Windows Setting window to connect the Odyssey + controllers. This appearently caused the controllers to have two references each in Windows. 

I removed the controllers from the Bluetooth connection settings and use thesetup from the Mixed Reality Portal. After all of this the behaviors are as expected and work fine. 
Again Thanks to all
